server/organisation_integration_test.go

```diff
@@ -0,0 +1,230 @@
+
+	// Act.
+	s.Router.ServeHTTP(w, req)
+
+	// Assert.
+	require.Equal(s.T(), expectedStatusCode, w.Code)
+
+	// Deep Assert.
+	var organisation datastore.Organisation
+	parseResponse(s.T(), w.Result(), &organisation)
+
+	org, err := s.DB.OrganisationRepo().FetchOrganisationByID(context.Background(), organisation.UID)
+	require.NoError(s.T(), err)
+	require.Equal(s.T(), "new_org", org.Name)
+}
+
+func (s *OrganisationIntegrationTestSuite) Test_CreateOrganisation_EmptyOrganisationName() {
+	expectedStatusCode := http.StatusBadRequest
+
+	body := strings.NewReader(`{"name":""}`)
+	// Arrange.
+	url := "/api/v1/organisations"
+	req := createRequest(http.MethodPost, url, body)
+	w := httptest.NewRecorder()
+
+	// Act.
+	s.Router.ServeHTTP(w, req)
+
+	// Assert.
+	require.Equal(s.T(), expectedStatusCode, w.Code)
+}
+
+func (s *OrganisationIntegrationTestSuite) Test_UpdateOrganisation_EmptyOrganisationName() {
+	expectedStatusCode := http.StatusBadRequest
+
+	uid := uuid.NewString()
+	_, err := testdb.SeedOrganisation(s.DB, uid, "", "new_org")
+	require.NoError(s.T(), err)
+
+	body := strings.NewReader(`{"name":""}`)
+	// Arrange.
+	url := fmt.Sprintf("/api/v1/organisations/%s", uid)
+	req := createRequest(http.MethodPut, url, body)
+	w := httptest.NewRecorder()
+
+	// Act.
+	s.Router.ServeHTTP(w, req)
+
+	// Assert.
+	require.Equal(s.T(), expectedStatusCode, w.Code)
+}
+
+func (s *OrganisationIntegrationTestSuite) Test_UpdateOrganisation() {
+	expectedStatusCode := http.StatusAccepted
+
+	uid := uuid.NewString()
+	_, err := testdb.SeedOrganisation(s.DB, uid, "", "new_org")
+	require.NoError(s.T(), err)
+
+	body := strings.NewReader(`{"name":"update_org"}`)
+
+	// Arrange.
+	url := fmt.Sprintf("/api/v1/organisations/%s", uid)
+	req := createRequest(http.MethodPut, url, body)
+	w := httptest.NewRecorder()
+
+	// Act.
+	s.Router.ServeHTTP(w, req)
+
+	// Assert.
+	require.Equal(s.T(), expectedStatusCode, w.Code)
+
+	org, err := s.DB.OrganisationRepo().FetchOrganisationByID(context.Background(), uid)
+	require.NoError(s.T(), err)
+	require.Equal(s.T(), "update_org", org.Name)
+}
+
+func (s *OrganisationIntegrationTestSuite) Test_GetOrganisation() {
+	expectedStatusCode := http.StatusOK
+
+	uid := uuid.NewString()
+	_, err := testdb.SeedOrganisation(s.DB, uid, "", "new_org")
+	require.NoError(s.T(), err)
+
+	// Arrange.
+	url := fmt.Sprintf("/api/v1/organisations/%s", uid)
+	req := createRequest(http.MethodGet, url, nil)
+	w := httptest.NewRecorder()
+
+	// Act.
+	s.Router.ServeHTTP(w, req)
+
+	// Assert.
+	require.Equal(s.T(), expectedStatusCode, w.Code)
+
+	// Deep Assert.
+	var organisation datastore.Organisation
+	parseResponse(s.T(), w.Result(), &organisation)
+
+	org, err := s.DB.OrganisationRepo().FetchOrganisationByID(context.Background(), uid)
+	require.NoError(s.T(), err)
+	require.Equal(s.T(), "new_org", org.Name)
+	require.Equal(s.T(), "new_org", organisation.Name)
+}
+
+func (s *OrganisationIntegrationTestSuite) Test_GetOrganisations() {
```

Comment:
@danvixent Can we refactor this to look like this - https://github.com/frain-dev/convoy/blob/b377ed70bb7ccb8af69c67c0d4c66fc3a6de1321/server/application_integration_test.go#L429

Reply:
@Subomi i'm not sure i see the difference here.

Reply:
The latter uses `testdb.SeedMultipleEndpoints` to seed multiple endpoints at once that use a randomly generated number of organisations. @danvixent

Reply:
We need something like `testdb.SeedMultipleOrganisations`

Reply:
Noted